larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java

```diff
@@ -188,17 +194,91 @@ public void delete(Object value, Object storage) throws EvalException {
         vault.delete(value, storage);
     }
 
+    @StarlarkMethod(
+        name = "sign",
+        doc = "signs a payload with given keyArn",
+        parameters = {
+            @Param(
+                name = "keyId",
+                doc = "key ARN to sign with",
+                allowedTypes = {
+                    @ParamType(type = String.class)
+                }
+            ),
+            @Param(
+                name = "message",
+                doc = "the payload to sign",
+                allowedTypes = {
+                    @ParamType(type = String.class)
+                }
+            ),
+            @Param(
+                name = "algorithm",
+                doc = "the algorithm to use to sign, valid choices are: RSASSA_PSS_SHA_256, RSASSA_PSS_SHA_384, " +
+                    "RSASSA_PSS_SHA_512",
+                allowedTypes = {
+                    @ParamType(type = String.class)
```

Comment:
I wonder if there's a way to restrict it

Reply:
we could probably use an enum in a smart way, but i added a validate function so it ends up being the same